operator/rpcoperator.go

```diff
@@ -60,63 +68,86 @@ func (o *RPCOperator) SubscribeProofs(addr common.Address) (ProofSub, error) {
 	o.mtx.Lock()
 	defer o.mtx.Unlock()
 	log.WithField("who", addr.Hex()).Debug("Subscribed to proofs")
-	sub, ok := o.subs[addr]
-	if !ok {
-		o.subscribe(addr)
-		sub = o.subs[addr]
-	}
-	return *sub, nil
+	sub := o.subscribe(addr)
+	return sub, nil
 }
 
 // subscribe is an internal implementation detail and should not be called.
 func (o *RPCOperator) subscribe(addr common.Address) ProofSub {
-	o.subs[addr] = newProofSub()
-	return *o.subs[addr]
+	sub := *newProofSub()
+	if _, ok := o.subs[addr]; !ok {
+		o.subs[addr] = new(BufferedProofSubs)
+	} else {
+		if dp := o.subs[addr].latestDP; dp != nil {
+			sub.deposits <- *dp
+		}
+		if bp := o.subs[addr].latestBP; bp != nil {
+			sub.balances <- *bp
+		}
+	}
+	o.subs[addr].subs = append(o.subs[addr].subs, sub)
+	return sub
 }
 
 func (o *RPCOperator) PushDepositProof(proof tee.DepositProof) {
 	o.mtx.Lock()
 	defer o.mtx.Unlock()
 	who := proof.Balance.Account
-	sub, ok := o.subs[who]
+	bufsub, ok := o.subs[who]
 	if !ok {
 		log.WithField("who", who.Hex()).Debug("Received DP without subscription - buffering")
-		o.subscribe(who)
-		sub = o.subs[who]
+		bufsub = new(BufferedProofSubs)
+		o.subs[who] = bufsub
 	}
-	// Clear out old buffered proof.
-	select {
-	case <-sub.deposits:
-	default:
-	}
-	// Write new proof in.
-	select {
-	case <-sub.quit:
-		delete(o.subs, who)
-	case sub.deposits <- proof:
+	o.subs[who].latestDP = &proof
```

Comment:
`bufsub` is a pointer, so you could do
```suggestion
	bufsub.latestDP = &proof
```
here, no?

Reply:
Good point.